app/assets/javascripts/discourse/app/components/glimmer-site-header.gjs

```diff
@@ -119,33 +123,24 @@ export default class GlimmerSiteHeader extends Component {
     }
 
     this._headerWrap = document.querySelector(".d-header-wrap");
+    this._mainOutletWrapper = document.querySelector("#main-outlet-wrapper");
     if (this._headerWrap) {
-      schedule("afterRender", () => {
-        this.headerElement = this._headerWrap.querySelector("header.d-header");
-        this.updateHeaderOffset();
-        document.documentElement.style.setProperty(
-          "--header-top",
-          `${this.headerElement.offsetTop}px`
-        );
-      });
+      this._recalculateHeaderOffset();
 
-      window.addEventListener("scroll", this._onScroll, {
+      window.addEventListener("scroll", this._recalculateHeaderOffset, {
         passive: true,
       });
 
+      this.appEvents.on("page:changed", this, this._recalculateHeaderOffset);
```

Comment:
Above/below header elements can be added/removed depending on the route, it's good to recalc.

Reply:
They can also be added/removed in other situations... but it's hard to know how we should handle that efficiently...

Could/should we wrap an element around all the PluginOutlets, and then use a ResizeObserver? 🤔

Reply:
Use a ResizeObserver on all PluginOutlets? I'm not sure, sounds like a lot of observers.

Reply:
No, I mean add a single wrapper element around everything which sits above the main-outlet. Then add a single ResizeObserver to that.

(but... maybe that wrapper element would cause layout issues... I dunno)

Reply:
Ah I see. Maybe we could add the observer to `.discourse-root`? Its height _should_ change when elements are added/removed, or when the element's height changes, anyway.

Reply:
Oh yes, good idea!

Reply:
Ok, I added a refactor @davidtaylorhq. We had a resize observer on the header wrap element, I remove that and replaced it with one for `.discourse-root`. It should cover all cases. 

Note that here we also have a scroll listener. That one fires a lot more often than the resize observer, and we didn't have a debouncer for it, so I added one. 

Also noticed one more thing. We have a second scroll listener on this file to add a `docked` class to the body. I don't think we are using the `docked` class on the body anymore, and in fact the scroll listener doesn't seem to be working properly. Will address this in a separate PR though.